### DISCUSSION TITLE
How to filter point cloud data by voxel selected?

### DISCUSSION THREAD

**Discussion by plsbstn** [isl-org/Open3D#5357] [Q&A]
Hi everyone 👋

I'm wondering how it's possible to select points from a PointCloud object by using specific voxels from a VoxelGrid object?
Thanks to certain methods, it's very easy to create VoxelGrid or Octree from a PointCloud.
But how can I pick points which are inside a voxel with index (x,y,z) like voxel.get_points_within(self) or pcd.select_by_voxel(voxel).

My scope is to filter point cloud data with respect to its topology neighborhood, represented by adjacent voxels, e.g.

```
voxel_grid = o3d.geometry.VoxelGrid.create_from_point_cloud(pcd, voxel_size=voxel_size)
indices = np.stack(list(vox.grid_index for vox in voxel_grid.get_voxels()))
for idx, vox in enumerate(indices):
            adjacent_voxels = create_adjacent_voxel(vox) //kind of meshgrid function to generate possible adjacent voxel triplets
            for each in indices[~idx]:
                        num_neighbor_voxels += len([(i,v) for i,v in enumerate(each==adjacent_voxels) if (v[0] and v[1] and v[2])]) //compare and count
            connected_neighbors_graph[idx] = num_neighbor_voxels //store in dict
dict((k, v) for k, v in connected_neighbors_graph.items() if v <10) //select those voxel with less than 10 connected neighbors
--> use this k (key) to select and eliminate points inside.
```

Any information to solve this issue will be highly appreciated 😀
Thanks!

**Comment by qixuema** [comment]
@plsbstn Hi, have you found a solution yet?
